src/core/syscalls/business_logic_syscall_handler.rs

```diff
@@ -36,8 +46,13 @@ impl SyscallHandler for BusinessLogicSyscallHandler {
         vm: &VirtualMachine,
         syscall_ptr: Relocatable,
     ) -> Result<(), SyscallHandlerError> {
-        let SyscallRequest::EmitEvent(request) =
-            self._read_and_validate_syscall_request("emit_event", vm, syscall_ptr)?;
+        let request = if let SyscallRequest::EmitEvent(request) =
+            self._read_and_validate_syscall_request("emit_event", vm, syscall_ptr)?
+        {
+            request
+        } else {
+            return Err(SyscallHandlerError::ExpectedEmitEventStruct);
+        };
```

Comment:
I think we should handle this with a match statement as has been done on other PRs, the idea would be to keep a standard way to manage this case.

Reply:
mmm, i don't think so.
In this cases, we are calling the `self._read_and_validate_syscall_request` method with a hardcoded string ( in this case "emit_event"). So I don't see the point in doing a match since this always returns an`EmitEventStruct`.
Actually, we have to add this if statement because of how we are implementing the code, but it is impossible to reach the `Err(SyscallHandlerError::ExpectedEmitEventStruct)`

Reply:
it suggested it as i considered it more clean, but you are right it is not necessary in this cases.